app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -852,10 +861,6 @@ public void onResumeSessionSafe() {
 
         registerFormEntryReceiver();
         restorePriorStates();
-
-        if (mFormController != null) {
-            mFormController.setPendingCalloutFormIndex(null);
```

Comment:
I have moved this to `onActivityResultSessionSafe`. This is done in order to take care of this specific scenario - 

1. User doesn't have camera permissions and he clicks on 'Take Picture' button
2. We trigger permission dialoge to get user's permission
3. User gives us the permission in response to which we call out to the system camera and set the call out index to say n.
4. onResume gets called on FormEntryActivity due to return from Permission dialog which resets the index back to null 
5. User takes the picture but since the call out index is set to null now the image doesn't get attached to the image capture question.